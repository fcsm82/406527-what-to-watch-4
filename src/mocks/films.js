import {getRandomElement, generateUniqueId} from "../utils/utils.js";

const FILMS_NUMBER = 8;

const TITLES = [
  `Fantastic Beasts: The Crimes of Grindelwald`,
  `Bohemian Rhapsody`,
  `Macbeth`,
  `Aviator`,
  `We need to talk about Kevin`,
  `What We Do in the Shadows`,
  `Revenant`,
  `Johnny English`,
  `Shutter Island`,
  `Pulp Fiction`,
  `No Country for Old Men`,
  `Snatch`,
  `Moonrise Kingdom`,
  `Seven Years in Tibet`,
  `Midnight Special`,
  `War of the Worlds`,
  `Dardjeeling Limited`,
  `Orlando`,
  `Mindhunter`,
  `Midnight Special`
];

const POSTERS = [
  `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  `bohemian-rhapsody.jpg`,
  `macbeth.jpg`,
  `aviator.jpg`,
  `we-need-to-talk-about-kevin.jpg`,
  `what-we-do-in-the-shadows.jpg`,
  `revenant.jpg`,
  `johnny-english.jpg`,
  `shutter-island.jpg`,
  `pulp-fiction.jpg`,
  `no-country-for-old-men.jpg`,
  `snatch.jpg`,
  `moonrise-kingdom.jpg`,
  `seven-years-in-tibet.jpg`,
  `midnight-special.jpg`,
  `war-of-the-worlds.jpg`,
  `dardjeeling-limited.jpg`,
  `orlando.jpg`,
  `mindhunter.jpg`,
  `midnight-special.jpg`
];

export const TOP_FILM = {
  title: `The Grand Budapest Hotel`,
  poster: `the-grand-budapest-hotel-poster.jpg`,
  genre: `Drama`,
  releaseDate: 2014
};

export const createFilms = () => {
  let films = [];
  for (let i = 0; i < FILMS_NUMBER; i++) {
    films.push({
      id: generateUniqueId(),
      title: getRandomElement(TITLES),
      poster: getRandomElement(POSTERS),
    });
  }
  return films;
};
