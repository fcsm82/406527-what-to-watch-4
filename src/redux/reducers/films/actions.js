export const Action = {
  CHANGE_FILTER: `Change filter`,
  SHOW_MORE: `Show more films`,
};

export const Creator = {
  filterFilms: (filter) => {
    return {
      type: Action.CHANGE_FILTER,
      payload: filter
    };
  },

  showMore: () => {
    return {
      type: Action.SHOW_MORE,
    };
  }
};
