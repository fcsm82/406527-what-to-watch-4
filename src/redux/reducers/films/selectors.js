// import {createSelector} from "reselect";
import NameSpace from "../name-space.js";
// import {getUniqueGenres} from "../../../utils/utils.js";
// import {getFilms} from "../data/selectors.js";

// export const getFilters = createSelector(
//     getFilms,
//     (films) => getUniqueGenres(films)
// );

export const getActiveFilter = (state) => {
  return state[NameSpace.FILMS].activeFilter;
};

export const getShownFilms = (state) => {
  return state[NameSpace.FILMS].shownFilms;
};

export const getView = (state) => {
  return state[NameSpace.FILMS].view;
};
