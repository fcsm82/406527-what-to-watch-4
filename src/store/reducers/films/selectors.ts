
import NameSpace from "../name-space";

export const getActiveFilter = (state) => {
  return state[NameSpace.FILMS].activeFilter;
};

export const getShownFilms = (state) => {
  return state[NameSpace.FILMS].shownFilms;
};
