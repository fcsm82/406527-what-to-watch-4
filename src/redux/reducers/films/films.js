
import {Action} from "../../actions.js";
// import {View} from "../../../components/app/app.jsx";
import {FILTER} from "../../../consts.js";

// import {getPromoFilm, getFilms} from "../data/selectors.js";

const initialState = {
  // view: View.LIST,
  shownFilms: 8,
  activeFilter: FILTER.ALL,
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case Action.CHANGE_FILTER:
      const filter = action.payload;

      return Object.assign({}, state, {
        activeFilter: filter,
        shownFilms: 8
      });

      // case Action.CHANGE_VIEW:
      //   return Object.assign({}, state, {
      //     view: View.DETAILS,
      //     activeFilm: action.payload
      //   });

      // case Action.CHANGE_VIEW_TO_SIGN_IN:
      //   return Object.assign({}, state, {
      //     view: View.SIGN_IN,
      //   });

    case Action.RENDER_FILMS:
      return Object.assign({}, state, {
        shownFilms: state.shownFilms + 8
      });

      // case Action.PLAY_VIDEO:
      //   return Object.assign({}, state, {
      //     view: View.VIDEO
      //   });

      // case Action.EXIT_VIDEO:
      //   return Object.assign({}, state, {
      //     view: View.LIST
      //   });

    default:
      return state;
  }
};

export {reducer};
