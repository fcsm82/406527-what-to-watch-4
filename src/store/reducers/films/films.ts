
import {Action} from "./actions";
import {ALL_GENRES} from "../../../consts";

const FilmsCount = {
  ON_START: 8,
  SHOW_MORE: 8
};

const initialState = {
  shownFilms: FilmsCount.ON_START,
  activeFilter: ALL_GENRES,
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case Action.CHANGE_FILTER:
      const filter = action.payload;

      return Object.assign({}, state, {
        activeFilter: filter,
        shownFilms: FilmsCount.ON_START
      });

    case Action.SHOW_MORE:
      return Object.assign({}, state, {
        shownFilms: state.shownFilms + FilmsCount.SHOW_MORE
      });

    default:
      return state;
  }
};

export {reducer};
