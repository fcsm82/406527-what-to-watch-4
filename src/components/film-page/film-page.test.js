import React from "react";
import {Router} from "react-router-dom";

import renderer from "react-test-renderer";
import FilmPage from "./film-page.jsx";
import history from "../../history.js";

import {film} from "../../mocks/films.js";


it(`Render FilmPage`, () => {
  const tree = renderer
    .create(
        <Router history={history}>
          <FilmPage
            film = {film}
            onPlayClick = {() => {}}
          />

        </Router>
        , {
          createNodeMock: () => {
            return {};
          }
        }).toJSON();

  expect(tree).toMatchSnapshot();
});
