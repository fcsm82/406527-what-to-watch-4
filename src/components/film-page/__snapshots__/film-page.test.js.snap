// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should FilmPage render correctly 1`] = `
Array [
  <div
    className="visually-hidden"
  >
    <svg
      xmlns="http://www.w3.org/2000/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <symbol
        viewBox="0 0 19 20"
      >
        <title>
          +
        </title>
        <g
          fill="none"
          fillRule="evenodd"
          strokeWidth="1"
        >
          <polygon
            fill="#EEE5B5"
            points="10.778 11.288 10.778 19.553 8.4165 19.553 8.4165 11.288 0.62793 11.288 0.62793 8.9268 8.4165 8.9268 8.4165 0.66211 10.778 0.66211 10.778 8.9268 18.566 8.9268 18.566 11.288"
          />
        </g>
      </symbol>
      <symbol
        fill="#FFF9D9"
        viewBox="0 0 27 27"
      >
        <path
          clipRule="evenodd"
          d="M23.8571 0H16V3.14286H23.8571V11H27V3.14286V0H23.8571Z"
          fillOpacity="0.7"
          fillRule="evenodd"
        />
        <path
          clipRule="evenodd"
          d="M27 23.8571V16H23.8571V23.8571H16V27H23.8571H27L27 23.8571Z"
          fillOpacity="0.7"
          fillRule="evenodd"
        />
        <path
          clipRule="evenodd"
          d="m0 3.1429v7.8571h3.1429v-7.8571h7.8571v-3.1429h-11v3.1429z"
          fillOpacity="0.7"
          fillRule="evenodd"
        />
        <path
          clipRule="evenodd"
          d="M3.14286 27H11V23.8571H3.14286L3.14286 16H0L0 23.8571V27H3.14286Z"
          fillOpacity="0.7"
          fillRule="evenodd"
        />
      </symbol>
      <symbol
        viewBox="0 0 18 14"
      >
        <path
          clipRule="evenodd"
          d="M2.40513 5.35353L6.1818 8.90902L15.5807 0L18 2.80485L6.18935 14L0 8.17346L2.40513 5.35353Z"
          fill="#EEE5B5"
          fillRule="evenodd"
        />
      </symbol>
      <symbol
        viewBox="0 0 14 21"
      >
        <g
          fill="#EEE5B5"
          fillRule="evenodd"
          strokeWidth="1"
        >
          <polygon
            fillRule="nonzero"
            points="0 -1.1191e-13 4 -1.1191e-13 4 21 0 21"
          />
          <polygon
            fillRule="nonzero"
            points="10 -1.1191e-13 14 -1.1191e-13 14 21 10 21"
          />
        </g>
      </symbol>
      <symbol
        id="play-s"
        viewBox="0 0 19 19"
      >
        <path
          clipRule="evenodd"
          d="M0 0L19 9.5L0 19V0Z"
          fill="#EEE5B5"
          fillRule="evenodd"
        />
      </symbol>
    </svg>
  </div>,
  <section
    className="movie-card movie-card--full"
    style={
      Object {
        "background": "#A6B7AC",
      }
    }
  >
    <div
      className="movie-card__hero"
    >
      <div
        className="movie-card__bg"
      >
        <img
          alt="Gangs of new york"
          src="https://htmlacademy-react-3.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg"
        />
      </div>
      <h1
        className="visually-hidden"
      >
        WTW
      </h1>
      <header
        className="page-header movie-card__head"
      >
        <div
          className="logo"
        >
          <a
            className="logo__link"
            href="/"
            onClick={[Function]}
          >
            <span
              className="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              className="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              className="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <div
          className="user-block"
        >
          <a
            href="/login"
            onClick={[Function]}
            style={
              Object {
                "color": "inherit",
                "cursor": "pointer",
                "textDecoration": "none",
              }
            }
          >
            Sign In
          </a>
        </div>
      </header>
      <div
        className="movie-card__wrap"
      >
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Gangs of new york
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Crime
            </span>
            <span
              className="movie-card__year"
            >
              2002
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <a
              className="btn btn--play movie-card__button"
              href="/player/1"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </a>
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="movie-card__wrap movie-card__translate-top"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster movie-card__poster--big"
        >
          <img
            alt="https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Gangs_of_New_York_Poster.jpg poster"
            height="327"
            src="https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Gangs_of_New_York_Poster.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <nav
            className="movie-nav movie-card__nav"
          >
            <ul
              className="movie-nav__list"
            >
              <li
                className="movie-nav__item movie-nav__item--active"
              >
                <a
                  className="movie-nav__link"
                  href="#"
                  onClick={[Function]}
                >
                  Overview
                </a>
              </li>
              <li
                className="movie-nav__item "
              >
                <a
                  className="movie-nav__link"
                  href="#"
                  onClick={[Function]}
                >
                  Details
                </a>
              </li>
              <li
                className="movie-nav__item "
              >
                <a
                  className="movie-nav__link"
                  href="#"
                  onClick={[Function]}
                >
                  Reviews
                </a>
              </li>
            </ul>
          </nav>
          <div
            className="movie-rating"
          >
            <div
              className="movie-rating__score"
            >
              8.8
            </div>
            <p
              className="movie-rating__meta"
            >
              <span
                className="movie-rating__level"
              >
                Very Good
              </span>
              <span
                className="movie-rating__count"
              >
                370881 ratings
              </span>
            </p>
          </div>
          <div
            className="movie-card__text"
          >
            <p>
              In 1862, Amsterdam Vallon returns to the Five Points area of New York City seeking revenge against Bill the Butcher, his father's killer.
            </p>
            <p
              className="movie-card__director"
            >
              <strong>
                Director: 
                Martin Scorsese
              </strong>
            </p>
            <p
              className="movie-card__starring"
            >
              <strong>
                Starring: 
                Leonardo DiCaprio, Cameron Diaz, Daniel Day-Lewis
                 and other
              </strong>
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog catalog--like-this"
    >
      <h2
        className="catalog__title"
      >
        More like this
      </h2>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/films/3"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
            >
              <video
                className="player__video"
                muted={true}
                poster="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/pulp-fiction.jpg"
                type="video/webm"
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/3"
              onClick={[Function]}
            >
              Pulp Fiction
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/films/7"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
            >
              <video
                className="player__video"
                muted={true}
                poster="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/Once_Upon_a_Time_in_America.jpg"
                type="video/webm"
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/7"
              onClick={[Function]}
            >
              Once Upon a Time in America
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/films/20"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
            >
              <video
                className="player__video"
                muted={true}
                poster="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/legend.jpg"
                type="video/webm"
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/20"
              onClick={[Function]}
            >
              Legend
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/films/21"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
            >
              <video
                className="player__video"
                muted={true}
                poster="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/no-country-for-old-men.jpg"
                type="video/webm"
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/21"
              onClick={[Function]}
            >
              No Country for Old Men
            </a>
          </h3>
        </article>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          © 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
